### ISSUE TITLE
Replace Enyo by vue.js

### ISSUE THREAD

**Issue by bzg** [llaske/sugarizer#257]
Sugarizer relies on the [Enyo JS](enyojs.com) framework but does not use anything from Enyo that makes it mandatory.

Replacing Enyo by [vue.js](https://vuejs.org/) would clarify and modernize the code base, while attracting more developers.

**Comment by usmanmuhd** [comment]
@llaske I am interested to work on this. Could you please provide me some pointers on where to get started with?

**Comment by ksraj123** [comment]
@llaske I would like to work on this as a part of my gsoc proposal. Before replacing enyo with vue in the core platform, Porting some activities first which use enyo would be a starting point I think.

The activities which currently use enyo framework are -
- Tank Operation
- Foodchain
- Abecedarium
- Tam Tam Micro
- Video Viewer
- Last One Loses

I would like a start working on porting a few of these. Which one among these would be a priority?
Requesting suggestions.

**Comment by llaske** [comment]
@ksraj123 Thanks for this proposal.
BTW I don't see any good reason to migrate these activities:
- No benefit for users: the activity will not work better
- No benefit for developers: there is no maintenance issue today and dependency to Enyo framework didn't not prevent recent improvements on these activities

So a best way to prepare migration is to invest in Vue.js for new activities. Not to rewrite old activities.
Conversely to Gtk2/Python2 issue in Sugar, using a deprecated Enyo version has no impact on Sugarizer deployment.

**Comment by quozl** [comment]
Well said.  We're not porting to Python 3 because we want to, but because we have no other way to benefit the users.

**Comment by ksraj123** [comment]
@llaske I agree with you. It sounds reasonable. As Enyo is a comparatively old framework there are not as many resources for us (student developers) to learn more about it or refer to if something does not work, as there are for vue. I made some contributions to some of the enyo activities I mentioned recently and faced these problems myself, seeing an open issue on this I thought maybe others are facing this issue as well. But I agree with you on porting will not make the old activities work better and I guess changes to the core platform are not that often to justify porting. So I agree with you on investing in vue for future activities. Thanks.

**Comment by llaske** [comment]
For information, it's a WIP here https://github.com/llaske/sugarizer/tree/feature/v2

**Comment by llaske** [comment]
Now merged on dev branch

**Comment by bzg** [comment]
Thanks!

**Comment by bzg** [comment]
🍸
